dash-pipeline/bmv2/dash_pipeline.p4

```diff
@@ -231,6 +235,98 @@ control dash_ingress(
         const default_action = deny;
     }
 
+    action set_meter_policy_attrs(@Sai[type="sai_ip_addr_family_t", isresourcetype="true"] bit<32> ip_addr_family) {
+        if (ip_addr_family == 0) /* SAI_IP_ADDR_FAMILY_IPV4 */ {
+            if (meta.is_overlay_ip_v6 == 1) {
+                meta.dropped = true;
+            }
+        } else {
+            if (meta.is_overlay_ip_v6 == 0) {
+                meta.dropped = true;
+            }
+        }
+    }
+
+    @name("meter_policy|dash_meter")
+    @Sai[isobject="true"]
+    table meter_policy {
+        key = {
+            meta.meter_policy_id : exact @name("meta.meter_policy_id:meter_policy_id");
+        }
+        actions = {
+            set_meter_policy_attrs();
+        }
+    }
+
+    action meter_rule_action(bit<16> meter_class) {
+        meta.policy_meter_class = meter_class;
+    }
+
+    @name("meter_rule|dash_meter")
+    @Sai[isobject="true"]
+    table meter_rule {
+        key = {
+            meta.meter_policy_id: exact @name("meta.meter_policy_id:meter_policy_id") @Sai[type="sai_object_id_t", isresourcetype="true", objects="METER_POLICY"];
+            hdr.ipv4.dst_addr : ternary @name("hdr.ipv4.dst_addr:dip");
+        }
+
+     actions = {
+            meter_rule_action;
+            @defaultonly deny;
+        }
+        const default_action = deny;
+    }
+
+    action meter_bucket_action(
+            @Sai[type="sai_uint32_t", isreadonly="true"] bit<32> outbound_bytes_counter,
+            @Sai[type="sai_uint32_t", isreadonly="true"] bit<32> inbound_bytes_counter) {
+        // read only counters for SAI api generation
+    }
+
+    @name("meter_bucket|dash_meter")
+    @Sai[isobject="true"]
+    table meter_bucket {
+        key = {
+            meta.eni_id: exact @name("meta.eni_id:eni_id");
+            meta.meter_class: exact @name("meta.meter_class:meter_class");
```

Comment:
what is not clear is if meter_class across all the types globally unique or each policy, route, mapping have their own scope of meter_class values? It is very important to be specified some-where. I did not see it mentioned in HLD. If scope of bit<16> meter_class is unique within scope of each policy, route, mapping separately then, have we not lost the context here while incrementing count here that will be reported.

Reply:
The scope of meter class-id is global and is managed above the SAI layer. The meter class-id usage and examples are mentioned in the HLD